packages/compiler-cli/ngcc/test/integration/ngcc_spec.ts

```diff
@@ -467,6 +467,58 @@ runInEachFileSystem(() => {
              .not.toThrow();
        });
 
+    it('should support inline UMD/CommonJS exports declarations', () => {
+      // Setup an Angular entry-point in UMD module format has an inline exports declaration
+      // referenced by an NgModule.
+      loadTestFiles([
+        {
+          name: _('/node_modules/test-package/package.json'),
+          contents: '{"name": "test-package", "main": "./index.js", "typings": "./index.d.ts"}'
+        },
+        {
+          name: _('/node_modules/test-package/index.js'),
+          contents: `
+          (function (global, factory) {
+            typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
+            typeof define === 'function' && define.amd ? define('test', ['exports', 'core'], factory) :
+            (factory(global.test, global.core));
+          }(this, (function (exports, core) { 'use strict';
+            exports.FooModule = /** @class */ (function () {
+              function FooModule() {}
+              FooModule = __decorate([
+                  core.NgModule({declarations: [exports.FooDirective]})
```

Comment:
You can make this more challenging when this is a reference to an inline declaration of an expression.
```suggestion
                  core.NgModule({declarations: exports.declarations})
```

and then have

```ts
exports.declarations = [exports.FooDirective];
```

this was the pattern that broke after the initial fix.

Reply:
Great! Will do.